plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/SyntheticColumnHandleBuilder.java

```diff
@@ -13,27 +13,46 @@
  */
 package io.trino.plugin.jdbc;
 
+import java.util.OptionalInt;
+
 import static com.google.common.base.Splitter.fixedLength;
 import static com.google.common.base.Verify.verify;
 
 public class SyntheticColumnHandleBuilder
```

Comment:
Two IF statements have been added to the logic in this class, making it maybe harder for the reader to understand what's going on? Usually in these situations I was trying to use the combined factory+executor/command pattern, that is to have a factory that would do the IF's and return different implementations with a different logic. Then you could name the classes accordingly:
-  NoLimitSynthetic...
- TruncatingSynthetic...
- LegacySynthetic...

Reply:
too many classes for not a lot of benefit imo.

do you think code comments would help?